Added prop 'showClearButton'

jomla97 opened the pull request.

khaiql commented:
Why do you have to check for the prop? There is no definition for `showClearButton` in https://github.com/jomla97/react-native-progressive-input/blob/507fc2a0ca923939dc4700f965feacacb0ee3abc/index.js#L13

jomla97 commented:
Sorry, I was in a bit of a rush because I needed the component without the clearButton.. I changed the prop name to something a little more logical that fits in with the rest of the clearButton props. Also added the PropTypes.
